sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receipt/_claims_models.py

```diff
@@ -0,0 +1,156 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Models for application claims."""
+
+from typing import Any, Dict, Optional, Union
+
+
+class LedgerEntryClaim:
+    """
+    LedgerEntryClaim represents an Application Claim derived from ledger entry data.
+    """
+
+    def __init__(
+        self,
+        protocol: str,
+        collectionId: str,
+        contents: str,
+        secretKey: str,
+    ):
+        """
+        :keyword protocol: The protocol used to compute the claim.
+        :paramtype protocol: str
+
+        :keyword collectionId: The collection ID of the ledger entry.
+        :paramtype collectionId: str
+
+        :keyword contents: The contents of the ledger entry.
+        :paramtype contents: str
+
+        :keyword secretKey: The secret key used to compute the claim digest.
+        :paramtype secretKey: str
+        """
+        self.protocol = protocol
+        self.collectionId = collectionId
+        self.contents = contents
+        self.secretKey = secretKey
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, LedgerEntryClaim):
+            return NotImplemented
+
+        return (
+            self.protocol == other.protocol
+            and self.collectionId == other.collectionId
+            and self.contents == other.contents
+            and self.secretKey == other.secretKey
+        )
+
+    @classmethod
+    def from_dict(cls, ledger_entry_claim_dict: Dict[str, Any]):
+        """Create a new instance of this class from a dictionary."""
+
+        return cls(
```

Comment:
Since the keys are named the same, you can do `cls(**ledger_entry_claim_dict)` if you want to be cool :smile:

Reply:
True, addressed. Thanks for the tip!